### PR TITLE
#415 import process of national appeal chamber judgments

### PR DESCRIPTION
**Pull request by madryk**

### CODE REVIEW

File: saos-batch/src/main/java/pl/edu/icm/saos/batch/core/NacjImportJobConfiguration.java [hunk 1]
```diff
@@ -68,4 +91,32 @@ protected Step nacJudgmentImportDownloadStep() {
             .listener(nacjImportDownloadStepExecutionListener)
             .build();
     }
+    
+    
+    
+    @Bean
+    public Job ctJudgmentImportProcessJob() {
```
Comment (lukdumi) [inline saos-batch/src/main/java/pl/edu/icm/saos/batch/core/NacjImportJobConfiguration.java]: typo: should be nacJud....

File: saos-import/src/main/java/pl/edu/icm/saos/importer/notapi/nationalappealchamber/judgment/process/NacjImportProcessStepExecutionListener.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    @Override
+    public void beforeStep(StepExecution stepExecution) {
+
+        judgmentObjectDeleter.deleteJudgmentsWithoutRawSourceJudgment(NationalAppealChamberJudgment.class, RawSourceNacJudgment.class);
+
+    }
+
+    @Override
+    public ExitStatus afterStep(StepExecution stepExecution) {
+        return null;
```
Comment (lukdumi) [inline saos-import/src/main/java/pl/edu/icm/saos/importer/notapi/nationalappealchamber/judgment/process/NacjImportProcessStepExecutionListener.java]: Why not <code>return ExitStatus.COMPLETED</code>?

File: saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/constitutionaltribunal/judgment/process/SourceCtJudgmentExtractorTest.java [hunk 3]
```diff
@@ -194,56 +192,13 @@ public void extractPublicationDate() {
...
-        Judge judgeJanNowak = new Judge(janNowak, JudgeRole.PRESIDING_JUDGE, JudgeRole.REPORTING_JUDGE);
-        Judge judgeAdamKowalski = new Judge(adamKowalski);
-        when(judgeConverter.convertJudge(janNowak, Lists.newArrayList(JudgeRole.PRESIDING_JUDGE, JudgeRole.REPORTING_JUDGE), correctionList)).thenReturn(judgeJanNowak);
-        when(judgeConverter.convertJudge(adamKowalski, Lists.newArrayList(), correctionList)).thenReturn(judgeAdamKowalski);
-        when(judgeConverter.convertJudge(wrongName, Lists.newArrayList(), correctionList)).thenReturn(null);
-        
-        
-        
-        // execute
+        Judge judge = new Judge("Jan Kowalski");
```
Comment (lukdumi) [inline saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/constitutionaltribunal/judgment/process/SourceCtJudgmentExtractorTest.java]: You'd better create a list (of judges) object here...

File: saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/constitutionaltribunal/judgment/process/SourceCtJudgmentExtractorTest.java [hunk 4]
```diff
@@ -194,56 +192,13 @@ public void extractPublicationDate() {
...
-        for (Judge judge : judges) {
-            sJudgment.getJudges().contains(judge.getName());
-            if (judge.getName().equals(sourceScJudge1.getName())) {
-                assertThat(judge.getSpecialRoles(), Matchers.containsInAnyOrder(JudgeRole.PRESIDING_JUDGE, JudgeRole.REPORTING_JUDGE));
-            } else {
-                assertThat(judge.getSpecialRoles(), Matchers.containsInAnyOrder(sourceScJudge2.getSpecialRoles().toArray()));
-            }
-        }
-        
+        assertEquals(1, judges.size());
```
Comment (lukdumi) [inline saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/constitutionaltribunal/judgment/process/SourceCtJudgmentExtractorTest.java]: ... and here assert if the returned list is the same object as expected
assertTrue(returnedJudges == judges)

File: saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/nationalappealchamber/judgment/process/SourceNacJudgmentExtractorTest.java [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+    
+    @Test
+    public void createNewJudgment() {
+        NationalAppealChamberJudgment nacJudgment = judgmentExtractor.createNewJudgment();
+        assertNotNull(nacJudgment);
+    }
+    
+    @Test
+    public void extractCourtCases() {
+
```
Comment (lukdumi) [inline saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/nationalappealchamber/judgment/process/SourceNacJudgmentExtractorTest.java]: given/ assert/ execute

File: saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/nationalappealchamber/judgment/process/SourceNacJudgmentExtractorTest.java [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+        DateTime publicationDate = judgmentExtractor.extractPublicationDate(sJudgment, correctionList);
+        
+        assertNull(publicationDate);
+        
+    }
+    
+    @Test
+    public void extractJudges() {
+        
+        Judge judge = new Judge("Jan Kowalski");
```
Comment (lukdumi) [inline saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/nationalappealchamber/judgment/process/SourceNacJudgmentExtractorTest.java]: The same trick with ImmutableList would be much easier and undoubted.

File: saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/supremecourt/judgment/process/SourceScJudgmentExtractorTest.java [hunk 7]
```diff
@@ -222,60 +220,13 @@ public void extractPublicationDate() {
...
-        Judge judgeAdamKowalski = new Judge(adamKowalski);
-        when(judgeConverter.convertJudge(janNowak, Lists.newArrayList(JudgeRole.PRESIDING_JUDGE, JudgeRole.REPORTING_JUDGE), correctionList)).thenReturn(judgeJanNowak);
-        when(judgeConverter.convertJudge(adamKowalski, Lists.newArrayList(), correctionList)).thenReturn(judgeAdamKowalski);
-        when(judgeConverter.convertJudge(wrongName, Lists.newArrayList(), correctionList)).thenReturn(null);
-        
-        
-        
-        // execute
+        Judge judge = new Judge("Jan Kowalski");
+        when(sourceJudgeExtractorHelper.extractJudges(sJudgment, correctionList)).thenReturn(Lists.newArrayList(judge));
```
Comment (lukdumi) [inline saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/supremecourt/judgment/process/SourceScJudgmentExtractorTest.java]: Same here, immutable + assert of object identity.

File: saos-batch/src/main/java/pl/edu/icm/saos/batch/core/NacjImportJobConfiguration.java (hunk 1 above)
Comment (madryk) [inline saos-batch/src/main/java/pl/edu/icm/saos/batch/core/NacjImportJobConfiguration.java]: :+1:

File: saos-import/src/main/java/pl/edu/icm/saos/importer/notapi/nationalappealchamber/judgment/process/NacjImportProcessStepExecutionListener.java (hunk 2 above)
Comment (madryk) [inline saos-import/src/main/java/pl/edu/icm/saos/importer/notapi/nationalappealchamber/judgment/process/NacjImportProcessStepExecutionListener.java]: Ok. I will return it.

File: saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/constitutionaltribunal/judgment/process/SourceCtJudgmentExtractorTest.java (hunk 3 above)
Comment (madryk) [inline saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/constitutionaltribunal/judgment/process/SourceCtJudgmentExtractorTest.java]: I will use `ImmutableList.of(judge)` - it's shorter

File: saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/constitutionaltribunal/judgment/process/SourceCtJudgmentExtractorTest.java (hunk 4 above)
Comment (madryk) [inline saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/constitutionaltribunal/judgment/process/SourceCtJudgmentExtractorTest.java]: Ok

File: saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/nationalappealchamber/judgment/process/SourceNacJudgmentExtractorTest.java (hunk 6 above)
Comment (madryk) [inline saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/nationalappealchamber/judgment/process/SourceNacJudgmentExtractorTest.java]: :+1:

File: saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/supremecourt/judgment/process/SourceScJudgmentExtractorTest.java (hunk 7 above)
Comment (madryk) [inline saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/supremecourt/judgment/process/SourceScJudgmentExtractorTest.java]: :+1:

File: saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/nationalappealchamber/judgment/process/SourceNacJudgmentExtractorTest.java (hunk 5 above)
Comment (madryk) [inline saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/nationalappealchamber/judgment/process/SourceNacJudgmentExtractorTest.java]: :+1:

**Pull request closed without merging**